app/react/Users/components/UserSidePanel.tsx

```diff
@@ -0,0 +1,267 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Icon } from 'UI';
+import { Translate } from 'app/I18N';
+import { ConfirmButton, SidePanel } from 'app/Layout';
+import { UserRole } from 'shared/types/userSchema';
+import { UserSchema } from 'shared/types/userType';
+import MultiSelect from 'app/Forms/components/MultiSelect';
+import { UserGroupSchema } from 'shared/types/userGroupType';
+import { PermissionsList } from 'app/Users/components/PermissionsList';
+
+export interface UserSidePanelProps {
+  user: UserSchema;
+  users: UserSchema[];
+  groups: UserGroupSchema[];
+  opened: boolean;
+  closePanel: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  onSave: (user: UserSchema) => void;
+  onDelete: (user: UserSchema) => void;
+  onReset2fa: (user: UserSchema) => void;
+  onResetPassword: (user: UserSchema) => void;
+}
+
+export const UserSidePanel = ({
+  user,
+  users,
+  groups,
+  opened,
+  closePanel,
+  onSave,
+  onDelete,
+  onReset2fa,
+  onResetPassword,
+}: UserSidePanelProps) => {
+  const [userToSave, setUserToSave] = useState(user);
+  const [permissionsModalOpened, setPermissionsModalOpened] = useState(false);
+  const [selectedGroups, setSelectedGroups] = useState(
+    user.groups ? user.groups.map(group => group._id!.toString()) : []
+  );
+
+  const { register, handleSubmit, errors } = useForm();
+
+  const availableGroups = groups;
+
+  const handleInputChange = (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => {
+    const { name, value } = event.currentTarget;
+    const updatedUser = { ...userToSave, [name]: value };
+    setUserToSave(updatedUser);
+  };
+
+  const onChangeHandler = (groupIds: string[]) => {
+    setSelectedGroups(groupIds);
+    const updatedGroups = groups
+      .filter(group => groupIds.includes(group._id as string))
+      .map(group => ({
+        _id: group._id!,
+        name: group.name,
+      }));
+    setUserToSave({ ...userToSave, groups: updatedGroups });
+  };
+
+  const isDuplicated = (nameVal: string) =>
+    !users.find(
+      existingUser =>
+        existingUser._id !== user._id &&
+        (existingUser.username?.trim().toLowerCase() === nameVal.trim().toLowerCase() ||
+          existingUser.email?.trim().toLowerCase() === nameVal.trim().toLowerCase())
+    );
+
+  const togglePermissionList = () => {
+    setPermissionsModalOpened(!permissionsModalOpened);
+  };
+
+  return (
+    <SidePanel open={opened}>
+      <div className="sidepanel-header">
+        <Translate>{`${user._id ? 'Edit' : 'Add'} User`}</Translate>
+      </div>
+      <div className="sidepanel-body">
+        <form id="userFrom" className="user-form" onSubmit={handleSubmit(() => onSave(userToSave))}>
+          <div id="email_field" className="form-group nested-selector">
+            <label>
+              <Translate>Email</Translate>
+            </label>
+            <input
```

Comment:
Did you try using a type='email' instead of validating with a regex?

Reply:
I didn't try, it's better, thanks.